backend/src/plugins/kube.ts

```diff
@@ -27,6 +28,13 @@ export default fp(async (fastify: FastifyInstance) => {
     fastify.log.error(e, 'Failed to retrieve current namespace');
   }
 
+  let saToken;
+  try {
+    saToken = await promiseWithTimeout(getSAToken(), 5000);
```

Comment:
This is an odd flow, waiting/racing for a timeout is not a bad practice... but your Promise literally has no way to go forward if you're not onCluster -- why the 5 second delay to starting up in dev?